atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -246,14 +246,27 @@ private CassandraClientPoolImpl(
     }
 
     private void tryInitialize() {
-        cassandra.cacheInitialCassandraHosts();
-
+        CassandraVerifierConfig verifierConfig = CassandraVerifierConfig.of(config, runtimeConfig.get());
         // for testability, mock/spy are bad at mockability of things called in constructors
         if (startupChecks == StartupChecks.RUN) {
-            runOneTimeStartupChecks();
+            ensureKeyspaceExists(verifierConfig);
```

Comment:
Needed because the client pool sets keyspace on clients, and for that to work, the keyspace must exist!

The implementation of ensureKeyspaceExists only ensures the keyspace exists on a single host. I decided against changing this to require all matching, because a) that would cause things to fail when Cassandra is rolling, and b) if, when constructing clients for the pools in CTV, we get a failure because the keyspace does not exist on a host, then that's an indicator that we actually have a splitbrain (we do have a best effort check that the schema versions in C* have stabilised). This will throw. In AsyncInitialisation, this will retry, and it's possible that the keyspace creator chooses the other cluster to create the keyspace (so it doesn't necessarily provide us with much signal), but does mean we don't _need_ to require an allMatch

 Note that we've separated ensuring the keyspace simply exists, and keeping it up to date. I figured we only need to keep the keyspace up to date on the cluster we actually intend to talk to! 

A few alternatives:
- Add everything to the pool, run the existing createKeyspaceAndUpToDate method, then validate and remove things from the pool. Pros - doesn't require CassVerifier changes, cons - a bit faffy ngl, but also a goal I have with this is that a pool is managed by CCPI iff it has been validated.
- Don't use CCPI in CassandraVerifier. Pros - breaks the dependency cycle, doesn't require splitting the methods up, cons - we'd create additional clients unnecessarily and need to either duplicate some CCPI logic or abstract it out, both being a little painful in CCPIs current state.

Reply:
Yeah when I originally wrote #6753 (and wrote about the alternative split) I was under the impression that I'd need to ensure keyspace on all the clusters (and by proxy, all hosts) which was a pain, but then later realised I could take the easy way out :)